Detectors/ITSMFT/ITS/reconstruction/include/ITSReconstruction/Clusterer.h

```diff
@@ -0,0 +1,54 @@
+/// \file Clusterer.h
+/// \brief Definition of the ITS cluster finder
+#ifndef ALICEO2_ITS_CLUSTERER_H
+#define ALICEO2_ITS_CLUSTERER_H
+
+#include <utility>
+#include <vector>
+
+#include "Rtypes.h"
+
+class TClonesArray;
+
+namespace o2
+{
+namespace ITS
+{
+  class PixelReader;
+  class Clusterer
+{
+ public:
+  Clusterer();
+  ~Clusterer() = default;
+
+  Clusterer(const Clusterer&) = delete;
+  Clusterer& operator=(const Clusterer&) = delete;
+
+  static void setPixelGeometry(Float_t px, Float_t pz, Float_t x0, Float_t z0) {
+    mPitchX=px; mPitchZ=pz; mX0=x0; mZ0=z0;
+  }
+  void process(PixelReader &r, TClonesArray &clusters);
+
+ private:
+  enum {kMaxRow=650}; //Anything larger than the real number of rows (512 for ALPIDE)
+  void initChip(UShort_t chipID, UShort_t row, UShort_t col, Int_t label);
+  void updateChip(UShort_t chipID, UShort_t row, UShort_t col, Int_t label);
+  void finishChip(TClonesArray &clusters);
+
+  Int_t mColumn1[kMaxRow+2];
+  Int_t mColumn2[kMaxRow+2];
+  Int_t *mCurr, *mPrev;
+  std::vector<std::vector<std::pair<UShort_t,UShort_t>>> mPreClusters;
```

Comment:
I don't think this will be efficient: during the clusterization pre-clusters are merged to each other. If the belonging of digits to given precluster is determined not by their being in container of the precluster but by their own index, the each time the preclusters are merged one should loop over all digits to fix the precluster index.